.circleci/config.yml

```diff
@@ -19,8 +19,8 @@ environment: &environment
 # Docker image
 docker: &docker
   docker:
-      # gcc-6.0, ruby-2.3, sudo, curl
-    - image: circleci/ruby:2.3
+      # ubuntu 22.04.2 LTS, build-essential, gcc-11
```

Comment:
the main reason for this upgrade is to fix the error 

```
./boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp:408:39: error: ‘check_linear_endpoints’ was not declared in this scope
```

that is related to a gcc-6 bug. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71546

Reply:
Gcc-6 does support C++14 so I don't think that bumping version of the compiler that is used for testing is the correct solution. Furthermore I'm not sure that this bug is the correct one.

A workaround would be to call the member function explicitly writing `this` keyword.

Reply:
I though it was a good opportunity to also upgrade the circleCI. Anyway, I am OK with the workaround and I remove the circleCI update from this PR. BTW do we want to do that upgrade in a different PR?

Reply:
Well it depends, do we want to catch such issues in the future?

Reply:
Good point. On the other hand we may miss issues that caught by newer versions. BTW, does it make sense to allow github actions run the full test-suite instead of the minimal one?

Reply:
Yes, and my guess is that more users compile with newer compiler and if that's true then testing it would be more important but I don't know really.

Regarding GH actions. I can't remember why have we decided to do it this way. Probably becasue running all of them would take a lot of time. But there may be some technical difficulties. I think the discussions should be in relevant PR. We should also check if there are some slots distributed across all of the Boost submodules because with long tests we'd take slots from other libraries.